### PR TITLE
Fixes gas containers exploading before destruction

### LINKED ISSUE #10198: [BUG] Crates explode when destroyed via damage but don't despawn until destroyed a second time
### There is no other issue reporting the same problem.

- [X] I have searched the existing issues

### Game Version

WIP 3921 but issue has been around for a while

### I was

Server

### OS

Linux

### What happened?

When a crate is damaged to the point of destruction, it gives a 'crate explodes' hud message and plays a sound effect. The crate is then returned to an apparently undamaged state, despawning without special effects the second time it is destroyed (not sure whether the gas contents are actually released the first or second time it's destroyed).

### What should have happened instead?

Crate despawn on initial explosion/destruction

### Steps to reproduce the issue

Spawn a crate
Shoot it, examining its damage state along the way

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MaxIsJoe**
fixes  #10198

CL: [New] Mappers and designers now have the ability to hook up functions to when an object/item get destroyed.
CL: [Fix] Fixed containers exploding before they are actually destroyed by explicitly asking if gas containers are meant to release their content due to excessive damage or when they are getting destroyed.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/GasContainer.cs [hunk 1]
```diff
@@ -66,6 +68,8 @@ public GasMix GasMix
 		[Tooltip("If true : Cargo will accept gases found within this container and can be sold.")]
 		public bool CargoSealApproved = false;
 
+		[SerializeField] private bool exploadOnTooMuchDamage = true;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/GasContainer.cs]:
```suggestion
		[SerializeField] private bool explodeOnTooMuchDamage = true;
```

Comment (corp-0) [reply UnityProject/Assets/Scripts/Objects/GasContainer.cs]: only change I see, looks good.

**Review by corp-0 — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Objects/GasContainer.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/GasContainer.cs]: done 👍

File: UnityProject/Assets/Scripts/Objects/GasContainer.cs [hunk 2]
```diff
@@ -68,7 +68,8 @@ public GasMix GasMix
 		[Tooltip("If true : Cargo will accept gases found within this container and can be sold.")]
 		public bool CargoSealApproved = false;
 
-		[SerializeField] private bool exploadOnTooMuchDamage = true;
+		[FormerlySerializedAs("exploadOnTooMuchDamage")] [SerializeField]
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/GasContainer.cs]: remove the formerlyserialized as. Re do your modification on the closet prefab instead.

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/GasContainer.cs]: annnd done 👍

**Review by corp-0 — APPROVED** [review]

**Pull request merged**